pandas/core/frame.py

```diff
@@ -528,9 +529,15 @@ def __init__(
                 values = cast_scalar_to_array(
                     (len(index), len(columns)), data, dtype=dtype
                 )
-                mgr = init_ndarray(
-                    values, index, columns, dtype=values.dtype, copy=False
-                )
+                if isinstance(values, list):
+                    # Case 1: values is a list of extension arrays
+                    dtype, _ = infer_dtype_from_array(values[0], pandas_dtype=True)
+                    mgr = arrays_to_mgr(values, columns, index, columns, dtype=dtype)
+                else:
+                    # Case 2: values is a numpy array
+                    mgr = init_ndarray(
+                        values, index, columns, dtype=values.dtype, copy=False
```

Comment:
please do not do this here

you can update init_array if needed

Reply:
I don't think we want this change in `init_ndarary`. Right now it turns a list of extension arrays into an ndarray, which we don't want here. That means we'd have the re-scan all the arrays in `values` to ensure that they're all EAs before calling `arrays_to_mgr`.

Where in a unique case here where we know that we have a list of EAs all with the same shape and type.

Reply:
Yeah it seems to me like changing `init_ndarray` would be require more complex handling, but I'm willing to make the change if that's what's requested. If so, I think I would benefit from a more concrete description of what you want, I'm not completely seeing it right now. Thanks!

Reply:
Please move this whole section to a separate function, This is make an already really long function REALLY long.

Reply:
This will need to stay here (https://github.com/pandas-dev/pandas/pull/34875#discussion_r446180430). We're going to have to send EA scalars down `ararys_to_mgr` and "ndarray" scalars down `init_ndarary`. I think that's unavoidable.